DOCS OBSDOCS-846 - Logging 5.7.12 Release Notes (#73272)
openshift/openshift-docs: modules/logging-release-notes-5-7-12.adoc

// module included in logging/logging-5-7-release-notes
:_mod-docs-content-type: REFERENCE
[id="logging-release-notes-5-7-12_{context}"]
= Logging 5.7.12

> This release includes link:https://access.redhat.com/errata/RHSA-2024:1508[Logging Bug Fix 5.7.12].

Reviewer: Just curious (as I do not know about this particular feature). Why doesn't this RN (and the previous RN) include "OpenShift" in title description but the earlier RNs do? 
![image](https://github.com/openshift/openshift-docs/assets/122639474/82a78de6-0960-4864-83a8-aeef2b01e931)

Author: This is probably because another writer worked on 5.7.10, also all the previous releases were drafted by me. :D I'll make the correction.

[id="logging-release-notes-5-7-12-bug-fixes"]
== Bug fixes

* Before this update, the Loki Operator checked whether the pods were running to decide if the `LokiStack` was ready. With this update, it also checks if the pods are ready, so the readiness of the `LokiStack` reflects the state of its components. (link:https://issues.redhat.com/browse/LOG-5172[LOG-5172])

> * Before this update, the Red Hat build pipeline didn't use the existing build details in Loki builds, leaving out information like revision, branch, and version. With this update, the Red Hat build pipeline now adds these inputs to the Loki builds, fixing the issue. (link:https://issues.redhat.com/browse/LOG-5202[LOG-5202])

Reviewer: ```suggestion
* Before this update, the Red Hat build pipeline didn't use the existing build details in Loki builds, leaving out information like revision, branch, and version. With this update, the Red Hat build pipeline now adds these details to the Loki builds, fixing the issue. (link:https://issues.redhat.com/browse/LOG-5202[LOG-5202])
```

Reviewer: Not sure if "inputs" is the same as the "existing details" mentioned earlier in the previous sentence, so not sure if they are interchangeable?

* Before this update, the configuration of the `ServiceMonitor` by Loki Operator could match many Kubernetes services, which led to Loki Operator's metrics being collected multiple times. With this update, the `ServiceMonitor` setup now only matches the dedicated metrics service. (link:https://issues.redhat.com/browse/LOG-5251[LOG-5251])

* Before this update, the build pipeline did not include linker flags for the build date, causing Loki builds to show empty strings for `buildDate` and `goVersion`. With this update, adding the missing linker flags in the build pipeline fixes the issue. (link:https://issues.redhat.com/browse/LOG-5275[LOG-5275])

> * Before this update, the Loki operator ServiceMonitor in openshift-operators-redhat used static token and CA files for authentication, causing errors in the Prometheus operator in User-Workload-Monitoring on such ServiceMonitor configuration. With this update, the Loki operator ServiceMonitor in openshift-operators-redhat now references a serviceaccount token secret via a LocalReference. This approach resolves the issue, allowing the User-Workload-Monitoring Prometheus operator to handle the Loki operator ServiceMonitor successfully, enabling Prometheus to scrape the Loki operator metrics. (link:https://issues.redhat.com/browse/LOG-5241[LOG-5241])

Reviewer: ```suggestion
* Before this update, the Loki operator ServiceMonitor in openshift-operators-redhat used static token and CA files for authentication, causing errors in the Prometheus operator in User-Workload-Monitoring on the ServiceMonitor configuration. With this update, the Loki operator ServiceMonitor in openshift-operators-redhat now references a serviceaccount token secret via a LocalReference. This approach resolves the issue, allowing the User-Workload-Monitoring Prometheus operator to handle the Loki operator ServiceMonitor successfully, enabling Prometheus to scrape the Loki operator metrics. (link:https://issues.redhat.com/browse/LOG-5241[LOG-5241])
```

Reviewer: Not sure if replacing "such" with "the" is correct, but do believe that "such" isn't correct?

Author: Made this change. Thanks!

// module included in logging/logging-5-7-release-notes
:_mod-docs-content-type: REFERENCE
[id="logging-release-notes-5-7-12_{context}"]
= Logging 5.7.12
This release includes link:https://access.redhat.com/errata/RHSA-2024:1508[OpenShift Logging Bug Fix 5.7.12].

[id="logging-release-notes-5-7-12-bug-fixes"]
== Bug fixes

* Before this update, the Loki Operator checked if the pods were running to decide if the `LokiStack` was ready. With this update, it also checks if the pods are ready, so that the readiness of the `LokiStack` reflects the state of its components.  (link:https://issues.redhat.com/browse/LOG-5172[LOG-5172])

> * Before this update, the Red Hat build pipeline didn't use the existing build details in Loki builds and omitted information such as revision, branch, and version. With this update, the Red Hat build pipeline now adds these details to the Loki builds, fixing the issue. (link:https://issues.redhat.com/browse/LOG-5202[LOG-5202])

Reviewer 2: Shouldn't use contractions. But I won't hold up the merge if this is the only thing

```suggestion
* Before this update, the Red Hat build pipeline did not use the existing build details in Loki builds and omitted information such as revision, branch, and version. With this update, the Red Hat build pipeline now adds these details to the Loki builds, fixing the issue. (link:https://issues.redhat.com/browse/LOG-5202[LOG-5202])
```
